src/string_util.cpp

```diff
@@ -113,6 +113,11 @@ namespace libtorrent
 
 	bool string_begins_no_case(char const* s1, char const* s2)
 	{
+		TORRENT_ASSERT(s1 != nullptr);
+		TORRENT_ASSERT(s2 != nullptr);
+
+		if (s1 == nullptr || s2 == nullptr) return false;
```

Comment:
this shouldn't happen, right? is it better to say false (and continue running with corrupt data) than to crash?